### PR TITLE
WIP: julia: update to 1.7.1.

### PR DESCRIPTION
**Pull request by non-Jedi**
Still need to finish building/testing locally and separating out into logical commits, but I thought I should put this here to make sure there's no duplicated effort since I'm not sure when I'll finish this up, and libblastrampoline was annoying to get to build correctly.

[ci skip] for building llvm.

#### Testing the changes
- I tested the changes in this PR: **YES**|**briefly**|**NO**

### CODE REVIEW

**Comment by kwshi** [comment]
thanks for doing the hard work on this, as a void julia user, i appreciate it! :heart:

**Comment by Piraty** [comment]
please take care of https://github.com/void-linux/void-packages/pull/35270

**Comment by Piraty** [comment]
libblastrampoline's tests require julia which would result in cyclic dependency, maybe put a comment ?

**Comment by Piraty** [comment]
also, https://github.com/JuliaLinearAlgebra/libblastrampoline/commit/39d859e6a848d191ba0895e95554b359a963e3cd is interesting (though appears not entirely correct for the static lib)

**Comment by non-Jedi** [comment]
> please take care of #35270

Looks like you linked the wrong issue/PR.

**Comment by non-Jedi** [comment]
Where I'm stuck at with this right now is that julia's build system seems to want to build all dependencies regardless of which `USE_SYSTEM_*` flags are set. This of course means that it tries to build libblastrampoline as a vendored dep which isn't successful without the flags I have set in the template for that package, so I haven't gotten julia 1.7 to successfully build at all yet. I'm going to ask around; feel like I must be doing something wrong.

**Comment by non-Jedi** [comment]
This appears to be some sort of issue with an interaction between the xbps-src system and julia's build system. Invoking `make` with the value of `make_build_args` causes only the desired libraries to be built, but when run from the `build` phase, it attempts to build all vedored dependencies entirely ignoring the `USE_SYSTEM_*` flags. Any ideas on this would be appreciated as I'm getting close to the end of my ability to debug this.

**Comment by kwshi** [comment]
I've attempted to take a look at this and tried to build it by hand (manually running `make` from the cloned source with correct arguments set).  Curiously, I seem to be running into a separate issue wherein the make build script is attempting but failing to detect/load/symlink the `libblastrampoline.so` library, even after I built and installed the `libblastrampoline` package included in this PR.

I'm not sure why--my best attempt at reading the Makefile to figure out what's happening seems to point to it using `libwhich` to locate libraries; when I try to manually run `libwhich -p libblastrampoline.so`, I get

```
failed to open library: /usr/lib/libblastrampoline.so: undefined symbol: f2c_sdot_64_
```

which suggests something additionally may be broken with the current state of the libblastrampoline package.  I doubt that's the same issue @non-Jedi you're facing with the `USE_*` variables getting ignored, but it seems to be yet another barrier in getting things to build correctly.  Have you encountered this?

File: srcpkgs/julia/template [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
  USE_SYSTEM_CURL=1 USE_SYSTEM_ZLIB=1 USE_SYSTEM_P7ZIP=1 USE_SYSTEM_LIBGIT2=1
- USE_SYSTEM_UTF8PROC=1
- UTF8PROC_INC='${XBPS_CROSS_BASE}/usr/include'"
+ USE_SYSTEM_UTF8PROC=1 USE_SYSTEM_LIBBLASTRAMPOLINE=1 USE_SYSTEM_CSL=1 USE_SYSTEM_LIBWHICH=1"
 make_install_args="$make_build_args"
 make_check_args="$make_build_args"
 make_check_target=testall
 conf_files="/etc/julia/startup.jl"
-hostmakedepends="pkg-config perl cmake python gcc-fortran patchelf which tar xz"
+hostmakedepends="pkg-config perl cmake python3 gcc-fortran patchelf which tar xz libwhich"
```
Comment (ahesford) [inline srcpkgs/julia/template:18]: I wonder if `libwhich` here is complicating the search for libraries in a way that triggers vendored builds of libraries you've installed in `makedepends`.

**Review by ahesford — COMMENTED** [review]
Unfortunately, I haven't had any time to look at the build problems here and am not sure when I'll have some spare time to dig in. The few points I make below are based just on reading the template and not actually trying to build it.

**Comment by Rabioli** [comment]
I was messing with julia on musl, I don't know what you guys are using but, at least for me, the void package appears to work but doesn't (it segfaults when trying to add new julia packages).

So I tried their official musl binary that doesn't even work either until I settled for compiling it myself which has given me the best result so far I haven't run into any issue with dependent on julia.

The relevancy into this issue is one, there is already a 1.7.2 release and soon to come out 1.8.0 and second could it be possible to export or reverse engineer the flags used by make "by default" so we can use them for the template?

I also attempted to make a template, but it just didn't work at all, like just changing the version and the checksums broke it all.

**Comment by Piraty** [comment]
oh, i meant to link https://github.com/void-linux/void-packages/issues/34387

**Comment by Piraty** [comment]
ping

**Comment by Piraty** [comment]
closing, see https://github.com/void-linux/void-packages/pull/39919

**Pull request closed without merging**